lib/presentation/pages/tip_resources_page.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_cached_pdfview/flutter_cached_pdfview.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+import 'package:infinite_horizons/presentation/atoms/atoms.dart';
+import 'package:infinite_horizons/presentation/atoms/separator_atom.dart';
+import 'package:infinite_horizons/presentation/molecules/molecules.dart';
+import 'package:youtube_player_flutter/youtube_player_flutter.dart';
+
+class TipResourcePage extends StatelessWidget {
+  const TipResourcePage({
+    required this.url,
+    super.key,
+  });
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    if (url.contains("pdf")) {
+      return Scaffold(
+        body: Column(
+          children: [
+            TopBarMolecule(
+              topBarType: TopBarType.back,
+              title: "Resource",
+              onTap: () => Navigator.pop(context),
+            ),
+            const SeparatorAtom(variant: SeparatorVariant.closeWidgets),
+            Expanded(child: pdfViewerMolecule(url, true, true, Colors.blue)),
+          ],
+        ),
+      );
+    } else if (url.contains("youtube") || url.contains("youtu.be")) {
+      return Stack(
+        children: [
+          GestureDetector(
+            onTap: () => Navigator.pop(context),
+            child: Container(
+              height: double.infinity,
+              color: Colors.transparent.withOpacity(0.5),
+            ),
+          ),
+          youtubePlayerMolecule(
+            url,
+            true,
+            true,
+            Colors.red,
+            Colors.red,
+            Colors.red,
+            const EdgeInsets.all(8.0),
+          ),
+        ],
+      );
+    } else {
+      return Scaffold(
+        body: Column(
+          children: [
+            TopBarMolecule(
+              topBarType: TopBarType.back,
+              title: "Resource",
+              onTap: () => Navigator.pop(context),
+            ),
+            const SeparatorAtom(variant: SeparatorVariant.closeWidgets),
+            Expanded(
+              child: InAppWebView(
+                initialUrlRequest: URLRequest(url: WebUri(url)),
+              ),
+            ),
+          ],
+        ),
```

Comment:
No need for else here, just return the widget. 
It will make the code more readable.

Reply:
I think this way is better for future devs because it shows a clear distinction between the three cases. Think about it once more and if you still want me to remove it I will.

Reply:
I understand where you are coming from.
But this way the code is more clean, developers should see the returns and take them into account while going through the code.

If you prefer something else you can create a state enum for each type and change it to a switch case